exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JsonRecordWriter.java

```diff
@@ -82,6 +84,9 @@ public void init(Map<String, String> writerOptions) throws IOException {
     Path fileName = new Path(location, prefix + "_" + index + "." + extension);
     try {
       stream = fs.create(fileName);
+      // set storage strategy for folder and file
+      storageStrategy.applyToFolder(fs, fileName.getParent());
```

Comment:
Does this handle the case where I ask to create "foo/bar/mumble/myFile" where "foo" exists but "bar" and "mumble" are both new? That is, does it set permissions on the full set of newly created folders? Should the `applyToFolder` also take on the `mkdirs` task so that it can create each folder in the path and set permissions on that folder?